client/signup/controller.js

```diff
@@ -296,6 +301,45 @@ export default {
 			context.store.dispatch( setSelectedSiteId( null ) );
 		}
 
+		if ( flowName === 'onboarding' || flowName === 'with-design-picker' ) {
+			context.primary = (
+				<ProvideExperimentData name="design_picker_after_onboarding">
+					{ ( isLoading, experimentAssignment ) => {
+						debug(
+							`design_picker_after_onboarding experiment variation: ${ experimentAssignment?.variationName }`
+						);
+
+						if ( isLoading ) {
+							debug( 'Waiting for design_picker_after_onboarding experiment status to load' );
+							return null;
+						}
+
+						const actualFlowName =
```

Comment:
Ideally we would be able to change the flow further in?
If so we could get it to no noticeable lag for participants.

Reply:
Due to how the signup framework is working, when changing the steps on the fly, we usually run into navigation problems. For example, in this case, I was trying to apply the experiment logic for all steps excluding the "user" `stepName` which is the one where the signup form shows. So it will cover pretty much all users that are loading Calypso for the first time in their browser (including experiment fetching). But this led to a buggy behavior identical with this one: https://github.com/Automattic/wp-calypso/issues/50273

Reply:
> we usually run into navigation problems

Here's one more after implementing the change 4ab9fd7 so we don't return `null` when the experiment is loading:
https://user-images.githubusercontent.com/14192054/116551084-363d1900-a900-11eb-86b5-ae085062d1d5.mov

Compared to this behavior, when pressing back after signing up in production:
- "user" step is displayed immediately
- I can manually advance to "domains" step instead of being automatically redirected
- There is a back button on "domains" step

Reply:
Thanks, @yansern, for the new round of [thorough testing](https://github.com/Automattic/wp-calypso/pull/52319#pullrequestreview-648264441)! 🚀